src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCIAuthorizationInterceptor.java

```diff
@@ -0,0 +1,30 @@
+package de.tum.in.www1.artemis.service.connectors.gitlabci;
+
+import java.io.IOException;
+
+import javax.validation.constraints.NotNull;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.service.connectors.gitlab.GitLabAuthorizationInterceptor;
+
+@Profile("gitlabci")
+@Component
+public class GitLabCIAuthorizationInterceptor implements ClientHttpRequestInterceptor {
```

Comment:
If I see this correctly the only difference between `GitLabAuthorizationInterceptor ` and `GitLabCIAuthorizationInterceptor` is the `@Profile` annotation. Can we find a better solution here to prevent duplicated code?

For example, we could add a class `AbstractGitLabAuthorizationInterceptor` which contains the two methods and two subclasses that only add the annotation. Would that work?

Reply:
I get your point. At the moment I don't have the time to implement and test your suggestion, but I will have a look at the weekend.